### PR TITLE
Replace rest-client with faraday

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

**Comment by drh-stanford** [comment]
The test coverage went down due to the new code added to modulator_job (client method).

**Comment by cbeer** [comment]
Maybe @tingulfsen can add tests that invokes `generate_xml`? It looks like the tests there are all marked pending.

**Comment by tingulfsen** [comment]
I will look into adding more tests. What is the reason for using faraday instead of rest-client?

**Comment by cbeer** [comment]
I believe the primary motivation is consistency; it was readily apparent in `Dor::WorkflowService` which used both rest-client and faraday within the same class. The interest in Faraday, in particular, is its support for persistent HTTP connections and more direct support for testing.

In this case, I hope the improved testing support will make it possible to write the pending modsulator tests here.

File: app/jobs/modsulator_job.rb [hunk 1]
```diff
@@ -256,4 +258,14 @@ def status_ok(dor_object)
...
     [1, 6, 7, 8, 9].include?(status)
   end
+
+  private
+
+  def client
+    Faraday.new do |f|
+      f.request :multipart
+      f.request :url_encoded
+      f.adapter :net_http
```
Comment (darrenleeweber) [inline app/jobs/modsulator_job.rb]: Can we look into using `http-client` here for better performance and persistent connections?
https://rubygems.org/gems/httpclient/
Compare with https://gist.github.com/darrenleeweber/89efe92d43449c07c729

**Comment by darrenleeweber** [comment]
Generally :+1: with a move to Faraday, depending on the actual adapter used.

File: app/jobs/modsulator_job.rb (hunk 1 above)
Comment (cbeer) [inline app/jobs/modsulator_job.rb]: Sure, looking into alternative HTTP adapters makes a lot of sense. Could we do that separate from switching to Faraday in the first place?

**Comment by cbeer** [comment]
I've added happy-path tests for `generate_xml`, and hope that @tingulfsen could back-fill the others (perhaps in a future pull request?)

**Comment by mejackreed** [comment]
:+1:

**Pull request merged**